catalog/imaging/imaging-transformer-nitf/src/test/java/org/codice/alliance/transformer/nitf/image/TestPreStoragePlugin.java

```diff
@@ -121,6 +124,71 @@ public void testNonNitfContent() throws PluginExecutionException, IOException {
         }
     }
 
+    @Test
+    public void testBuildDerivedImageTitle() {
+        String title = "Too Legit To Quit";
+        String qualifier = "qualifier";
+
+        String derivedFilename = nitfPreStoragePlugin.buildDerivedImageTitle(title, qualifier);
+
+        assertThat(StringUtils.containsNone(derivedFilename, FILESYSTEM_INVALID_CHARS), is(true));
+        assertThat(derivedFilename, is("qualifier-TooLegitToQuit.jpg"));
+    }
+
+    @Test
+    public void testBuildDerivedImageTitleIsNull() {
+        String title = null;
+        String qualifier = "QualifierOnly";
+
+        String derivedFilename = nitfPreStoragePlugin.buildDerivedImageTitle(title, qualifier);
+
+        assertThat(StringUtils.containsNone(derivedFilename, FILESYSTEM_INVALID_CHARS), is(true));
+        assertThat(derivedFilename, is("QualifierOnly.jpg"));
+    }
+    @Test
+    public void testBuildDerivedImageTitleIsEmpty() {
+        String title = "";
+        String qualifier = "QualifierOnly";
+
+        String derivedFilename = nitfPreStoragePlugin.buildDerivedImageTitle(title, qualifier);
+
+        assertThat(StringUtils.containsNone(derivedFilename, FILESYSTEM_INVALID_CHARS), is(true));
+        assertThat(derivedFilename, is("QualifierOnly.jpg"));
+    }
+
+    @Test
+    public void testBuildDerivedImageTitleWithInvalidChars() {
+        String title = "A bunch of _invalid_ characters! @#$%^&*()+-={}|[]<>?:";
+        String qualifier = "qualifier";
+
+        String derivedFilename = nitfPreStoragePlugin.buildDerivedImageTitle(title, qualifier);
+
+        assertThat(StringUtils.containsNone(derivedFilename, FILESYSTEM_INVALID_CHARS), is(true));
+        assertThat(derivedFilename, is("qualifier-Abunchof_invalid_characters.jpg"));
+    }
+
+    @Test
+    public void testbuildDerivedImageTitleAllInvalidChars() {
+        String title = "@#$%^&*()+-={}|[]<>?:";
+        String qualifier = "QualifierOnly";
+
+        String derivedFilename = nitfPreStoragePlugin.buildDerivedImageTitle(title, qualifier);
+
+        assert(StringUtils.containsNone(derivedFilename, FILESYSTEM_INVALID_CHARS));
+        assertThat(derivedFilename, is("QualifierOnly.jpg"));
+    }
+
+    @Test
+    public void testbuildDerivedImageTitleAllValidSpecialChars() {
```

Comment:
Updated to be a JUnit parameterized test. Will definitely give Spock a try next time around!